components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,618 @@
+                return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                        ErrorMessages.LOGOUT_SERVER_EXCEPTION.getMessage());
+            }
+            UserSessionManagementService userSessionManagementService =
+                    OpenIDConnectAuthenticatorDataHolder.getInstance()
+                            .getUserSessionManagementService();
+            userSessionManagementService.terminateSessionsByUserId(userId);
+            log.info("Sessions terminated for user Id: " + userId);
+            return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_OK,
+                    OIDCAuthenticatorConstants.BackchannelLogout.LOGOUT_SUCCESS);
+        } catch (SessionManagementException e) {
+            throw handleLogoutServerException(ErrorMessages.USER_SESSION_TERMINATION_FAILURE, e, userId);
+        }
+    }
+
+    private String getUserId(String tenantDomain, String sub, IdentityProvider identityProvider)
+            throws LogoutServerException {
+
+        try {
+            int tenantId = tenantDomain == null ? -1 : IdentityTenantUtil.getTenantId(tenantDomain);
+            String userId = UserSessionStore.getInstance()
+                    .getFederatedUserId(sub, tenantId, Integer.parseInt(identityProvider.getId()));
+            if (StringUtils.isBlank(userId)) {
+                throw handleLogoutServerException(ErrorMessages.RETRIEVING_USER_ID_FAILED, sub);
+            }
+            return userId;
+        } catch (UserSessionException e) {
+            throw handleLogoutServerException(ErrorMessages.RETRIEVING_USER_ID_FAILED, e, sub);
+        }
+    }
+
+    /**
+     * Validate the JWT token signature and the mandatory claim according to the OIDC specification.
+     *
+     * @param signedJWT
+     * @param identityProvider
+     * @return boolean value
+     * @throws LogoutClientException
+     * @throws LogoutServerException
+     */
+    private void validateLogoutToken(SignedJWT signedJWT, IdentityProvider identityProvider)
+            throws LogoutClientException, LogoutServerException {
+
+        try {
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+
+            if (!JWTSignatureValidationUtils.validateSignature(signedJWT, identityProvider)) {
+                throw new LogoutClientException(ErrorMessages.LOGOUT_TOKEN_SIGNATURE_VALIDATION_FAILED.getCode(),
+                        ErrorMessages.LOGOUT_TOKEN_SIGNATURE_VALIDATION_FAILED.getMessage());
+            }
+            validateAudience(claimsSet.getAudience(), identityProvider);
+            validateIat(claimsSet.getIssueTime());
+            validateEventClaim((JSONObject) claimsSet.getClaim(OIDCAuthenticatorConstants.Claim.EVENTS));
+            validateNonce(claimsSet);
+            if (!isSubClaimExists(claimsSet)) {
+                throw new LogoutClientException(ErrorMessages.LOGOUT_TOKEN_SUB_CLAIM_VALIDATION_FAILED.getCode(),
```

Comment:
throw the exception in the method itself

Reply:
Subject claim is optional one according to the spec https://openid.net/specs/openid-connect-backchannel-1_0.html
> sub
> OPTIONAL. Subject Identifier, as specified in Section 2 of [OpenID.Core].